packages/credential-registry/README.md

```diff
@@ -0,0 +1,74 @@
+# CredentialRegistry DataModel
+
+DataModel implementation of the **CredentialRegistry** schemas and definition.
+
+## Installation
+
+```sh
+npm install -D @datamodels/credential-registry
+```
+
+## Rationale
+
+The CredentialRegistry data-model holds the list of claimed [ClaimedCredentials](./schemas/ClaimedCredential.json) and the [VerifiableCredentials](./schemas/VerifiableCredential.json) issued by the user.
```

Comment:
I don't really understand the purpose of ClaimedCredentials.

It seems like an unnecessary step before the issuance of actually [VerifiableCredentials](https://www.w3.org/TR/vc-data-model/#credentials), which represent a collection of claims made(and signed) by an issuer, which in this case will be a DID.

Can you please elaborate on the expected flow of data/claims/signatures here? Perhaps we have different models in mind.

Reply:
Hi @mirceanis, here is how I'm thinking this:

When you think of centralized Issuers, it would be expected for them to issue the credentials for the users, even without the user's consent. But as a decentralized user, I would like to be able to claim things about my identity, and then ask Issuers to sign Verifiable Credentials for those claims. Maybe I would ask one issuer, maybe several of them. 

Also, as you point out, as a user, I want to have my private data, encrypted for my DID in the ClaimedCredential stream, and then have VerifiableCredentials to be issued for those claims without the plain data, but with a hash instead.

Still, I also think Issuers should be able create VerifiableCredentials directly, and use the credentialSubject "type" and "typeSchema" (or maybe more context entries, I need to check how other implementers do this) or additionalProperties (although others have commented we should avoid additionalProperties:true in Ceramic) to extend the possible claims. 

Please take a look at this demo videos to check the flow proposed for an app using this datamodel: https://docs.krebit.id/#/get_verified?id=claim-credentials

Thanks,
Alejandro

Reply:
> When you think of centralized Issuers, it would be expected for them to issue the credentials for the users, even without the user's consent. But as a decentralized user, I would like to be able to claim things about my identity, and then ask Issuers to sign Verifiable Credentials for those claims. Maybe I would ask one issuer, maybe several of them.

As a user I could also issue VerifiableCredentials about myself, there is no need for a different format here, nor a different stream.

> Also, as you point out, as a user, I want to have my private data, encrypted for my DID in the ClaimedCredential stream, and then have VerifiableCredentials to be issued for those claims without the plain data, but with a hash instead.

So the ClaimedCredential stream, and all the others are streams of encrypted data?

> Still, I also think Issuers should be able create VerifiableCredentials directly, and use the credentialSubject "type" and "typeSchema" (or maybe more context entries, I need to check how other implementers do this) or additionalProperties (although others have commented we should avoid additionalProperties:true in Ceramic) to extend the possible claims.

I was suggesting the `"additionalProperties": true` for credentialSubject because otherwise it would seem that the stream cannot truly host VerifiableCredentials, but only an incredibly tiny subset of VerifiableCredentials whose only claim is an ethereum address. You said these VerifiableCredentials should be able to express at least a hash of a ClaimedCredential, but the model you propose does not mention this hash in the schema of the `credentialSubject`.

Reply:
> As a user I could also issue VerifiableCredentials about myself, there is no need for a different format here, nor a different stream.
Thanks for pointing this out! So in other words, the schema I'm calling ClaimedCredential could be just a VerifiableCredential, where the Issuer is the same CredentialSubject, right?. I think that actually makes a lot of sense.

> So the ClaimedCredential stream, and all the others are streams of encrypted data?
Following the above comment, we could have 2 types of VCs on the did-store:
- VCs issued by the subject itself: if the properties on the credentialSubject are private, they should be encrypted to their DID
- VCs issued by other issuers: if the properties on the credentialSubject are private, they are stored as a hash 

That way the schema is more generic, and then a proposed flow (at least for my use case) would be:
1. A user creates a self-issued VC and stores it to its "issuedCredentials" definition record, with some encrypted credentialSubject fields.
2. The user shares the VC, evidence, and the plain values of the credentialSubject via some encrypted chat/access means to an Issuer 
3. The Issuer checks the data and issues a VC back to the user, with the private data as a hash in the CredentialSubject
4. Verifiers then could ask the user to share the plain data, and the issuer VC. They just need to verify that the plain data hash matches the issuer VC (and there would be no need to attach the evidence, thus preserving some privacy)

> I was suggesting the `"additionalProperties": true` for credentialSubject because otherwise it would seem that the stream cannot truly host VerifiableCredentials, but only an incredibly tiny subset of VerifiableCredentials whose only claim is an ethereum address. You said these VerifiableCredentials should be able to express at least a hash of a ClaimedCredential, but the model you propose does not mention this hash in the schema of the `credentialSubject`.
Actually, I'm seeing we already have the default "additionalProperties": true on the credentialSubject, so we're good there.

Reply:
> the schema I'm calling ClaimedCredential could be just a VerifiableCredential, where the Issuer is the same CredentialSubject, right?.

Isn't there a difference between a user issuing a credential about themselves vs taking a credential issued by someone else and publicly associating it with their DID in their DIDDataStore to make it discoverable?  The latter is what I thought the use case for ClaimedCredential was.  Issuers won't be able to write credentials they issue to the DIDDataStore of the user the credential is about - only users can write to their own DIDDataStore.  So if a user wants to have a list of credentials about themselves, regardless of who issued them, all viewable and discoverable as associated with their own DID, then they need to add those credentials to their DIDDataStore.

Reply:
@krebit there is a bit more to privacy than simply encrypting or hashing the claims. Think of hashing the "yes" or "no" answer to a vote. It would be trivial for someone to perform the same hash on the plain values and determine the value that was hashed. Even when the values are encrypted, there is a high correlation risk, since the stream data is public.

IMO, the protocol between the user and the various issuers should be entirely private, instead of using a public stream of claims that can later be signed by issuers.

Of course, there are legitimate use-cases for public claims and public credentials but whenever these claims or credentials are about Personal Identifiable Information, active measures should be applied to prevent them from becoming public, especially on a potentially permanent, public media such as the ceramic network.